### DISCUSSION TITLE
关于情绪值的设想

### DISCUSSION THREAD

**Discussion by HydroGest** [YesWeAreBot/YesImBot#29] [Ideas]
在 LLM 的返回的 JSON 里添加一个 `emotion` 键，类型为 number。

例如返回为 `5` 表示 +5 情绪值，`-10` 表示 -10 情绪值。

类似的，在提供给 AI 的提示词里也提示 AI 当前的 **累计情绪值**，让 AI 根据当前的情绪值自行决定回复的语气。

为了更真实地还原人类情绪变化， Athena 可以根据上次修改情绪值的时间和当前时间的距离调整情绪使其恢复到 **平静状态**，此时的情绪可以称为 **基准情绪**。例如上次 AI 和群友大吵一架，过了 24h 情绪逐渐恢复到正常情况。（也就是累计情绪值会趋向于向与基准情绪值的距离减小的方向变化）

**Comment by Touch-Night** [comment]
用哪个函数根据当前情绪、基准情绪和上次修改的时间来确定这次请求前的最新情绪值呢？要允许用户自定义情绪值随时间恢复的函数吗？
情绪有很多个维度，可能难以用单个值描述。

**Reply by HydroGest** [reply]
我设计了一个情绪波动模型，具体包含以下几个关键组成部分：

1. 基准情绪(s)：这是一个人在平静状态下的情绪基准线

2. 指数衰减项：`A * e^(-kt)`
   - 描述从初始情绪(e_0)逐渐回归到基准情绪的过程
   - k 是衰减率，决定回归速度
   - A 是初始偏离基准的幅度

3. 周期性波动：`B * sin(ωt)`
   - 模拟日常情绪的自然波动
   - ω 决定周期长度（如24小时一个周期）
   - B 决定波动幅度

4. 随机扰动项：
   - 模拟各种随机因素对情绪的影响
   - 幅度较小，但能反映现实生活中的不确定性

完整的数学模型可以表示为：
```
e(t) = s + A*e^(-kt) + B*sin(ωt) + ξ(t)
```
其中：
- e(t) 是 t 时刻的情绪值
- s 是基准情绪
- ξ(t) 是随机扰动项（可选）

模拟结果：

![Screenshot_20241117_180345.jpg](https://github.com/user-attachments/assets/52b493f4-0c94-4581-9da2-46a8b2b70572)

**Comment by Touch-Night** [comment]
使用多维向量来表示情绪，而不是单个值，如何呢？情绪值代表的含义有些不太明确

**Reply by HydroGest** [reply]
好主意
